### PR TITLE
SWATCH-297: Configure RHODS product in tag_profile

### PR DESCRIPTION
**Pull request by ksynvrit**
This is to address https://issues.redhat.com/browse/SWATCH-297

Add new tags for rhods in tagprofile yaml
Update ProdcutId in api-spec
Add new queryParam template for addons in metering worker yaml.

Testing Steps:

1. Setup profiles in your configuration with the following
```
DEV_MODE=true;SPRING_PROFILES_ACTIVE=openshift-metering-worker;PROM_URL=http://localhost:8082/api/metrics/v1/telemeter/api/v1;USER_USE_STUB=true
```
2. Add the following in MeteringJMXBean to test that 'rhods' would be accepted as a query.
```
http://localhost:9000/hawtio/jolokia/exec/org.candlepin.subscriptions.metering.jmx:name=meteringJmxBean,type=MeteringJmxBean/performCustomMeteringForAccount(java.lang.String,java.lang.String,java.lang.String,java.lang.Integer).

The parameters: 301721915996, rhods, 2022-10-31T19:00:00Z, 20160
```
3. Observe that Prometheus runs the query without issue.
```
[org.candlepin.subscriptions.metering.jmx.MeteringJmxBean] - rhods metering for 301721915996 against range [2022-10-17T19:00Z, 2022-10-31T19:00Z) triggered via JMX by anonymousUser
2022-10-07 16:15:41,987 [thread=http-nio-9000-exec-9] [INFO ] [org.candlepin.subscriptions.metering.service.prometheus.task.PrometheusMetricsTaskManager] - Queuing rhods Cores metric updates for account 301721915996.
2022-10-07 16:15:41,987 [thread=http-nio-9000-exec-9] [INFO ] [org.candlepin.subscriptions.metering.service.prometheus.task.PrometheusMetricsTaskManager] - Queuing rhods Cores metric update for account 301721915996 for range [2022-10-17T19:00Z, 2022-10-31T19:00Z)
2022-10-07 16:15:41,987 [thread=http-nio-9000-exec-9] [INFO ] [org.candlepin.subscriptions.metering.service.prometheus.task.PrometheusMetricsTaskManager] - ACCOUNT: 301721915996 TAG: rhods METRIC: Cores START: 2022-10-17T19:00Z END: 2022-10-31T19:00Z
2022-10-07 16:15:41,987 [thread=http-nio-9000-exec-9] [INFO ] [org.candlepin.subscriptions.metering.service.prometheus.task.PrometheusMetricsTaskManager] - Done queuing updates of rhods Cores metric
2022-10-07 16:15:41,988 [thread=pool-3-thread-2] [INFO ] [org.candlepin.subscriptions.metering.task.MetricsTask] - Running rhods Cores metrics update task for account: 301721915996
2022-10-07 16:15:41,993 [thread=pool-3-thread-2] [INFO ] [org.candlepin.subscriptions.metering.service.prometheus.PrometheusMeteringController] - Collecting metrics for account 301721915996: rhods Cores
2022-10-07 16:15:41,993 [thread=pool-3-thread-2] [INFO ] [org.candlepin.subscriptions.metering.service.prometheus.PrometheusService] - Fetching metrics from prometheus: 2022-10-17T20:00Z -> 2022-10-31T19:00Z [Step: 3600]
2022-10-07 16:15:42,277 [thread=pool-3-thread-2] [INFO ] [org.candlepin.subscriptions.metering.service.prometheus.PrometheusMeteringController] - Persisted 0 events for rhods Cores metrics.
2022-10-07 16:15:42,279 [thread=pool-3-thread-2] [INFO ] [org.candlepin.subscriptions.metering.task.MetricsTask] - rhods Cores metrics task complete.
```
You can play around and try to find an event from Prometheus using Thanos -> http://localhost:8082/api/metrics/v1/telemeter 
Below is a PROM query that I use as an template:
```
ocm_subscription_resource{resource_type="addon",resource_name="addon-open-data-hub",billing_model="marketplace"}
```

### CODE REVIEW

File: swatch-core/src/main/resources/tag_profile.yaml [hunk 1]
```diff
@@ -263,6 +273,19 @@ tagMetrics:
...
+   #RHODS metrics
+  - tag: rhods
+    metricId: redhat.com:openshift_data_science:cpu_hours
+    rhmMetricId: redhat.com:openshift_data_science:cpu_hours
+    awsDimension: cpu_hour
+    uom: CORES
+    billingWindow: MONTHLY
+    queryKey: addonSamples
+    queryParams:
+      resourceName: rhods
```
Comment (mstead) [inline swatch-core/src/main/resources/tag_profile.yaml:285]: Given our target PromQL query is supposed to be:
```
cluster:usage:workload:capacity_physical_cpu_hours
            * on(_id) group_right
            min_over_time(ocm_subscription_resource{resource_type="addon",resource_name='addon-open-data-hub', billing_model="marketplace", support=~"Premium|Standard|Self-Support|None"}[1h])
```

Setting the `resourceName` to `rhods` does not produce the correct query and we will get no results.

`resourceName` should be: `addon-open-data-hub`

**Review by mstead — CHANGES_REQUESTED** [review]
The PromQL query that is being run is incorrect.

**Review by mstead — COMMENTED** [review]
So based on the metric data I see in stage, the `ebs_account` label (account number) is not included in the metrics, therefore, the promql will not match any metrics currently.

For this to work against stage data, we will need to do at least one of the following:
1. Ask them to include the 'ebs_account' label in the prometheus metrics.
2. The orgId IS there (`external_organization` label), BUT we will need to have [Adjust the PromQL queries to use external_organization label](https://issues.redhat.com/browse/SWATCH-205) done before this will actually work.

**Comment by mstead** [comment]
@ksynvrit 

As per @kahowell :
> Let's just implement the other card first.
i.e. keep #1486 in draft mode until the other implemented

File: swatch-core/src/main/resources/tag_profile.yaml (hunk 1 above)
Comment (ksynvrit) [reply swatch-core/src/main/resources/tag_profile.yaml:285]: Will refactor

**Review by kahowell — APPROVED** [review]

**Pull request merged**